packages/@aws-cdk/aws-appsync/lib/graphqlapi-base.ts

```diff
@@ -0,0 +1,193 @@
+import { ITable } from '@aws-cdk/aws-dynamodb';
+import { IFunction } from '@aws-cdk/aws-lambda';
+import { CfnResource, IResource, Resource } from '@aws-cdk/core';
+import { DynamoDbDataSource, HttpDataSource, LambdaDataSource, NoneDataSource } from './data-source';
+
+/**
+ * Optional configuration for data sources
+ */
+export interface DataSourceOptions {
+  /**
+   * The name of the data source, overrides the id given by cdk
+   *
+   * @default - the id provisioned
```

Comment:
```suggestion
   * @default - the id provisioned
```

let's make this a little more helpful and say `Generated by ` ...

Reply:
wait but the name is just the id for now..